mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -252,6 +253,35 @@ private[ml] object TreeRegressorParams {
   final val supportedImpurities: Array[String] = Array("variance").map(_.toLowerCase)
 }
 
+private[ml] trait DecisionTreeRegressorParams extends DecisionTreeParams
+  with TreeRegressorParams with HasVarianceCol {
+
+  /** @group setParam */
+  def setVarianceCol(value: String): this.type = set(varianceCol, value)
+
+  /**
+   * Validates and transforms the input schema with the provided param map.
+   * @param schema input schema
+   * @param fitting whether this is in fitting
+   * @param featuresDataType SQL DataType for FeaturesType.
+   *                         E.g., [[org.apache.spark.mllib.linalg.VectorUDT]] for vector features.
+   * @return output schema
+   */
+  override protected def validateAndTransformSchema(
+      schema: StructType,
+      fitting: Boolean,
+      featuresDataType: DataType): StructType = {
+    SchemaUtils.checkColumnType(schema, $(featuresCol), featuresDataType)
```

Comment:
You could eliminate code by calling the parent's version of validateAndTransformSchema.